pkg/controllers/user/multiclusterapp/controller_multiclusterapp.go

```
@@ -133,40 +132,32 @@ func (m *MCAppController) sync(key string, mcapp *v3.MultiClusterApp) (runtime.O
 
 // createNamespaceAndApp creates the namespace for all workloads of the app, and then the app itself
 func (m *MCAppController) createNamespaceAndApp(t *v3.Target, mcapp *v3.MultiClusterApp, answerMap map[string]map[string]string, ann map[string]string,
-	set map[string]string, projectNS string, creatorID string, appNameInProject string, externalID string) (*v3.Target, *v3.MultiClusterApp, error) {
+	set map[string]string, projectNS string, creatorID string, externalID string) (*v3.Target, *v3.MultiClusterApp, error) {
 	var answerFound bool
-	targetNamespace := appNameInProject
 	// Create the target namespace first
 	// Adding the projectId as an annotation is necessary, else the API/UI and UI won't list any of the resources from this namespace
 	n := v1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        targetNamespace,
-			Labels:      map[string]string{projectIDFieldLabel: projectNS},
-			Annotations: map[string]string{projectIDFieldLabel: t.ProjectName, creatorIDAnn: creatorID},
+			GenerateName: mcapp.Name + "-",
+			Labels:       map[string]string{projectIDFieldLabel: projectNS},
+			Annotations:  map[string]string{projectIDFieldLabel: t.ProjectName, creatorIDAnn: creatorID},
 		},
 	}
-	_, err := m.namespaces.Get(targetNamespace, metav1.GetOptions{})
+	ns, err := m.namespaces.Create(&n)
```

Not checking if this already exists, if it does we can return the error